### DISCUSSION TITLE
Best way to find the index of column

### DISCUSSION THREAD

**Discussion by rtpsw** [ibis-project/ibis#5838] [Q&A] [answered]
What is the best way to find the index of a column within a table? More specifically, given a table `t : TableExpr` and one of its columns `c : *Column`, how to find the index of `c` in the list of columns of `t`?

Using Ibis 3.0.0, one thing I tried was:
```
cols = [col for col in t]
i = cols.index(c)
```
which doesn't work, apparently because of the overloaded `==` operator. At least in one session, I also found that `[col is c for col in t]` returned `False` for all.

I ended up doing:
```
col_names = t.schema().names
i = col_names.index(c.get_name())
```
which doesn't seem elegant. I'm also not sure whether Ibis guarantees that column names of a table are unique; if not, this code would be incorrect.

**Comment by cpcloud** [comment, marked as answer]
How about this? 

```
i = t.columns.index(c.get_name())
```

**Reply by rtpsw** [reply]
This is a shorter version of what I ended up doing, as noted above - this finds the index based on column names. Is this the best way? Is there a guarantee that column names of a table are unique?

**Reply by cpcloud** [reply]
> Is there a guarantee that column names of a table are unique?

Yes. You can't construct an expression with duplicate column names, because you can't construct a schema with duplicate names, and schemas are constructed for every table expression.

**Reply by cpcloud** [reply]
Also, in `[col is c for col in t]` hopefully *always* returns a `list` entirely of `False`s. `c` is a column expression and iterating over a table object yields column names, *not* column objects.
